web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,275 @@
+            setUsername('');
+            return;
+          }
+          throw new Error(`No SSO connectors found for user: ${username}`);
+        }
+        // If there isn't a remembered username, and there is only one matching connector, we take them straight to the IdP.
+        if (connectors.length === 1 && !rememberedUsername) {
+          onLoginWithSso(connectors[0]);
+          storageService.setRememberedSsoUsername(username);
+          setRememberedUsername(username);
+          return;
+        }
+        setRememberedUsername(username);
+        storageService.setRememberedSsoUsername(username);
+        return;
+      },
+      [username]
+    )
+  );
+
+  const onSubmitUsername = () => {
+    fetchMatchingConnectors(username.trim());
+  };
+
+  const onNotYou = () => {
+    storageService.clearRememberedSsoUsername();
+    setUsername('');
+    setRememberedUsername('');
+    setConnectors([]);
+  };
+
+  return (
+    <Flex flexDirection="column" alignItems="center" width="100%">
+      {fetchAttempt.status === 'error' && (
+        <Alerts.Danger>{fetchAttempt.statusText}</Alerts.Danger>
+      )}
+      {!rememberedUsername ? (
+        <>
+          <UsernamePrompt
+            onSubmitUsername={onSubmitUsername}
+            username={username}
+            setUsername={setUsername}
+            fetchAttempt={fetchAttempt}
+          />
+          <ButtonLink
+            mt={3}
+            onClick={onUseLocalLogin}
+            disabled={fetchAttempt.status === 'processing'}
+          >
+            Sign in another way
+          </ButtonLink>
+        </>
+      ) : (
+        <Text typography="h2" textAlign="center" mb={2}>
+          Welcome back, {username}
+        </Text>
+      )}
+      {connectors.length === 1 && (
+        <OneConnector
+          provider={connectors[0]}
+          onLoginWithSso={onLoginWithSso}
+          onNotYou={onNotYou}
+          fetchAttempt={fetchAttempt}
+        />
+      )}
+      {connectors.length > 1 && (
+        <MultipleConnectors
+          providers={connectors}
+          onLoginWithSso={onLoginWithSso}
+          onNotYou={onNotYou}
+          fetchAttempt={fetchAttempt}
+        />
+      )}
+    </Flex>
+  );
+}
+
+function UsernamePrompt({
+  onSubmitUsername,
+  username,
+  setUsername,
+  fetchAttempt,
+}: {
+  onSubmitUsername(): void;
+  username: string;
+  setUsername: (username: string) => void;
+  fetchAttempt: Attempt<void>;
+}) {
+  const inputRef = useRefAutoFocus<HTMLInputElement>({
+    shouldFocus: true,
+  });
+
+  return (
+    <Flex
+      as="form"
+      alignItems="center"
+      justifyContent="center"
+      flexDirection="column"
+      onSubmit={e => {
+        e.preventDefault();
+        onSubmitUsername();
+      }}
+      width="100%"
+    >
+      <Text typography="h3" mb={3}>
+        Enter your username to log in with SSO
+      </Text>
+      <Input
+        ref={inputRef}
+        value={username}
```

Comment:
should we validate username to be of email account type since it needs to be sso account and matched with the connector?